src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,161 @@
+package duke.storage;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+/**
+ * Stores the tasks and data provided by the user, so that when the program re-opens data can be restored.
+ */
+public class Storage {
+    //TODO create file method
+    //TODO error when tasklist is null for empty constructor
+    private static final String STORAGE_PATH = "data/duke.txt";
+    private static final File DATA_FILE = new File(STORAGE_PATH);
+
+    private String storagePath;
+    private File dataFile;
+
+    private TaskList taskList = new TaskList();
+
+    public Storage(String storagePath) {
+        this.storagePath = storagePath;
+        dataFile = new File(storagePath);
+    }
+
+    /**
+     * Loads the stored data into the program when it restarts.
+     *
+     * @return a list of stored tasks
+     * @throws IOException
+     */
+    public List<Task> load() throws IOException {
+        return txtToList(dataFile);
+    }
+
+    /**
+     * Converts a TaskList into a txt file to be stored.
+     *
+     * @param fileWriter
+     * @throws IOException
+     */
+    public void listToTxt(FileWriter fileWriter) throws IOException {
+        for (int i = 0; i < taskList.size(); i++) {
+            String currentTask = taskList.getTask(i).toString() + "\n";
+            fileWriter.write(currentTask);
+        }
+        fileWriter.close();
+    }
+
+    /**
+     * Converts a txt file into a list of tasks to be loaded when the program restarts.
+     *
+     * @param dataFile
+     * @return a list of stored tasks
+     * @throws IOException
+     */
+    public List<Task> txtToList(File dataFile) throws IOException {
+        List<Task> taskList = new ArrayList<>();
+        String currentTaskString = "";
+        BufferedReader reader = new BufferedReader(new FileReader(dataFile));
+
+        do {
+            currentTaskString = reader.readLine();
+            if (currentTaskString != null) {
+                Task currentTask = getTaskFromString(currentTaskString);
+                taskList.add(currentTask);
+            }
+        } while (currentTaskString != null);
+
+        return taskList;
+
+    }
+
+    private Task getTaskFromString(String taskString) {
+        String taskType = getTaskTypeFromString(taskString);
+        String taskName = getTaskNameFromString(taskString);
+        boolean taskStatus = getTaskStatusFromString(taskString);
```

Comment:
Perhaps it would be better to name this ```isDone``` (same as in the ```Task``` class) since it is a boolean?